### PR TITLE
Issue#1720 update org edit

### LINKED ISSUE #1720: Org - Profile update
## UI 
[Invision](https://projects.invisionapp.com/share/KFVPVJ87J49/#/screens/403392408)

## Acceptance criteria
- [ ] the forms are loaded with the pre-existing data 
- [ ] use org.addresses.main for the headquarters address & contact phone
- [ ] email -> org.email
- [ ] Bank - > org.bankAccounts (holder name blocked by #1583

### PR DESCRIPTION
**Pull request by jenna1k**
resolve #1720

### CODE REVIEW

File: apps/catalog/catalog/src/app/marketplace/title/view/view.component.ts [hunk 1]
```diff
@@ -91,10 +91,10 @@ export class MarketplaceMovieViewComponent implements OnInit {
     const { workType, totalRunTime, genres, originalLanguages } = movie.main;
     return [
       workType,
-      `${totalRunTime} min`,
+      typeof totalRunTime === 'number' ? `${totalRunTime} min` : 'TBC',
       genres.map(genre => getLabelBySlug('GENRES', genre)).join(', '),
       originalLanguages.map(language => language).join(', ')
-    ].join(' | ');
+    ].filter(Boolean).join(' | ');
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/marketplace/title/view/view.component.ts]: I would have done: `filter(value => !!value)`

File: libs/organization/src/lib/forms/organization-form/organization-form.component.html [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    </mat-card>
+    <mat-card>
+      <mat-card-header>
+        <mat-card-title>Company Logo</mat-card-title>
+        <mat-card-subtitle>Upload your Logo</mat-card-subtitle>
+      </mat-card-header>
+      <mat-card-content>
+        <drop-cropper [ratio]="1" storagePath="logo" formControlName="logo">
+          <h3 type>Logo</h3>
+            <p class="mat-body-2" dimensions>Recommanded : 512 x 512 px</p>
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/forms/organization-form/organization-form.component.html]: format on the same line as h3

File: libs/organization/src/lib/forms/organization-form/organization-form.component.html [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+          <input
+            matInput
+            placeholder="Fiscal number"
+            aria-label="fiscal number"
+            formControlName="fiscalNumber"
+          />
+        </mat-form-field>
+      </mat-card-content>
+    </mat-card>
+    <ng-container *ngFor="let control of bankAccounts.controls">
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/forms/organization-form/organization-form.component.html]: Add comments between your sections to make your code more readable

File: libs/organization/src/lib/forms/organization-form/organization-form.component.html [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        <mat-card-content>
+          <h5>bank account detail</h5>
+          <div fxLayout="row" fxLayoutAlign="space-between center" fxLayoutGap="15px">
+            <mat-form-field appearance="outline">
+              <mat-label>IBAN</mat-label>
+              <input matInput placeholder="IBAN" [formControl]="control.get('IBAN')" />
+            </mat-form-field>
+            <mat-form-field appearance="outline">
+              <mat-label>BIC or SWIFT</mat-label>
+              <input matInput placeholder="BIC or SWIFT" [formControl]="control.get('BIC')" />
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/forms/organization-form/organization-form.component.html]: I think youwant a mat select here instead.

File: libs/organization/src/lib/forms/organization-form/organization-form.component.html [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+              <input matInput placeholder="City" aria-label="city" [formControl]="control.get('address').get('city')" />
+            </mat-form-field>
+            <mat-form-field appearance="outline">
+              <mat-label>Zip code</mat-label>
+              <input matInput placeholder="Zip code" aria-label="zip code" [formControl]="control.get('address').get('zipCode')" />
+            </mat-form-field>
+          </div>
+          <mat-form-field appearance="outline">
+            <mat-label>Country</mat-label>
+            <input matInput placeholder="Country" aria-label="country" [formControl]="control.get('address').get('country')" />
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/forms/organization-form/organization-form.component.html]: use the form-country inside ui/form

File: libs/organization/src/lib/forms/organization-form/organization-form.component.scss [hunk 6]
```diff
@@ -2,16 +2,15 @@ mat-form-field {
...
   display: block;
   width: 100%;
 }
+
+mat-card {
+  margin-bottom: 15px;
+}
+
+h5 {
+  text-transform: uppercase;
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/forms/organization-form/organization-form.component.scss]: 👍

File: libs/organization/src/lib/pages/organization/organization.component.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import { OrganizationQuery } from '@blockframes/organization/+state/organization.query';
+import { OrganizationService } from '@blockframes/organization/+state/organization.service';
+import { Observable, Subscription } from 'rxjs';
+import { Organization } from '@blockframes/organization/+state/organization.model';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'organization-edit',
+  templateUrl: './organization.component.html',
+  styleUrls: ['./organization.component.scss']
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization/organization.component.ts]: add changeDetectionStrategy

File: libs/organization/src/lib/pages/organization/organization.component.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  constructor(
+    private query: OrganizationQuery,
+    private service: OrganizationService,
+    private snackBar: MatSnackBar,
+  ) { }
+
+  ngOnInit() {
+    this.sub = this.query
+      .selectActive()
+      .subscribe(org => this.organizationForm.patchValue(org));
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization/organization.component.ts]: Use `patchAllValue`. This is a custom method from FormEntity that make the FormList work as expected.

File: libs/organization/src/lib/pages/view/view.component.ts [hunk 9]
```diff
@@ -21,17 +25,20 @@ const   navLinks = [{
 export class OrganizationViewComponent implements OnInit {
   public organization$: Observable<Organization>;
   public previousPage: string;
-  public placeholderLogo = 'Organization_250.png';
   public navLinks = navLinks;
+  public organizationForm = new OrganizationForm(this.service);
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/view/view.component.ts]: You shouldn't pass a service to a Form but an organization entity instead. This means that `OrganizationForm` is not well defined. Update it please.

File: libs/organization/src/lib/forms/organization.form.ts [hunk 10]
```diff
@@ -71,3 +72,21 @@ function createOrganizationAddressesControls(addresses: Partial<AddressSet> = {}
 type OrganizationAddressesControl = ReturnType<typeof createOrganizationAddressesControls>
 
 
+function createOrganizationBankAccountFormControl(account: Partial<BankAccount>) {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/forms/organization.form.ts]: You want account to be optional

File: libs/organization/src/lib/forms/organization.form.ts [hunk 11]
```diff
@@ -71,3 +72,21 @@ function createOrganizationAddressesControls(addresses: Partial<AddressSet> = {}
...
+    BIC: new FormControl(entity.BIC),
+    name: new FormControl(entity.name),
+    holderName: new FormControl(entity.holderName)
+  }
+}
+
+export type OrganizationBankAccountFormControl = ReturnType<typeof createOrganizationBankAccountFormControl>;
+
+export class OrganizationBankAccountForm extends FormEntity<OrganizationBankAccountFormControl> {
+  constructor(account: Partial<BankAccount>) {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/forms/organization.form.ts]: You want account parameter to be optional

File: libs/organization/src/lib/pages/organization/organization.component.html [hunk 12]
```diff
@@ -0,0 +1,7 @@
+<organization-form [form]="organizationForm"></organization-form>
+<footer fxLayout="row" fxLayoutAlign="center center">
+  <button mat-flat-button color="accent" (click)="update()">
+    <mat-icon svgIcon="floppy"></mat-icon>
+    <span>Update</span>
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization/organization.component.html]: 👍

File: libs/organization/src/lib/pages/organization/organization.component.ts [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+  public sub: Subscription;
+  
+  constructor(
+    private query: OrganizationQuery,
+    private service: OrganizationService,
+    private snackBar: MatSnackBar,
+  ) { }
+
+  ngOnInit() {
+    this.sub = this.query
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization/organization.component.ts]: Use snapshot for none-collaboratif forms : 
```typescript
const organization = tihs.query.getActive();
this.form = new OrganizationForm(organization, this.service);
```

File: libs/utils/src/lib/form/validators/validators.ts [hunk 14]
```diff
@@ -67,8 +67,11 @@ export function validPercentage(control: FormControl): ValidationErrors {
 }
 
 /** Check if the `name` field of an Organization create form already exists as an ENS domain */
-export function UniqueOrgName(service: OrganizationService): AsyncValidatorFn {
+export function UniqueOrgName(service: OrganizationService, c: FormControl): AsyncValidatorFn {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/form/validators/validators.ts]: Give real name for your parameters : `c` => `***`
Like here, I don't know the difference between "c: FormControl" and "control: AbstractControl"

File: libs/organization/src/lib/pages/view/view.component.ts (hunk 9 above)
Comment (jenna1k) [reply libs/organization/src/lib/pages/view/view.component.ts]: issue created https://github.com/blockframes/blockframes/issues/1876

**Pull request merged**